src/main/java/de/zalando/aruha/nakadi/exceptions/NakadiException.java

```diff
@@ -26,4 +31,11 @@ public void setProblemMessage(final String problemMessage) {
         this.problemMessage = problemMessage;
     }
 
+    public Problem asProblem() {
+        final String detail = StringUtils.isNotBlank(problemMessage) ? problemMessage : getMessage();
+        return Problem.valueOf(getStatus(), detail);
+    }
+
+    protected abstract Response.StatusType getStatus();
```

Comment:
If we agree that all NakadiExceptions are mean to be used in a "controller", then I'm fine with this :+1: